### PR TITLE
14 create endpoint  test accounts/:msaid

### LINKED ISSUE #14: Create Endpoint & Test: /accounts/{msaId}
As a developer, we want to be able to get an Account.

## Acceptance criteria

- [ ] Create a new file for the endpoint
- [ ]  Endpoint should return the corresponding account to the given msaId
- [ ] Endpoint should use corresponding types defined in /dtos
- [ ] Create a test for the endpoint

### PR DESCRIPTION
**Pull request by claireolmstead**
closes #14 

## Changes
- create an endpoint for accounts/:msaid  to fetch accounts
- refactored so that each endpoint has its own controller and service
- added test for accounts endpoint
- refactored types to fit our needs better.

### CODE REVIEW

File: apps/api/src/controllers/accounts.controller.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   * @returns A promise that resolves to an array of AccountDTO objects representing the found account.
+   * @throws An error if the account cannot be found.
+   */
+  async getAccount(@Param('msaId') msaId: number): Promise<AccountResponse> {
+    try {
+      const account = await this.accountsService.getAccount(msaId);
+      return account;
+    } catch (error) {
+      this.logger.error(error);
+      throw new HttpException('Failed to find the account', HttpStatus.BAD_REQUEST);
```
Comment (mattheworris) [inline apps/api/src/controllers/accounts.controller.ts:60]: I think a throw here will result in a 500.
```suggestion
      return HttpException('Failed to find the account', HttpStatus.BAD_REQUEST);
```

File: README.md [hunk 2]
```diff
@@ -37,8 +37,8 @@ For example, here are some interactions that are provided by account-service:
 
    This will start Frequency, Redis and Account Service api/worker containers.
 
-3. Go to [BullUI](http://0.0.0.0:3000/queues/)  and check the graph service queue.
-4. Check the [Swagger](http://0.0.0.0:3000/api/docs/swagger) for API documentation.
+2. Go to [BullUI](http://0.0.0.0:3000/queues/)  and check the graph service queue.
+2. Check the [Swagger](http://0.0.0.0:3000/api/docs/swagger) for API documentation.
```
Comment (mattheworris) [inline README.md:41]:
```suggestion
3. Go to [BullUI](http://0.0.0.0:3000/queues/)  and check the graph service queue.
4. Check the [Swagger](http://0.0.0.0:3000/api/docs/swagger) for API documentation.
```

File: apps/api/src/controllers/accounts.controller.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  Logger,
+  Param,
+  HttpException,
+} from '@nestjs/common';
+import { ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { AccountsService } from '../services/accounts.service';
+import { AccountResponse } from '../../../../libs/common/src/dtos/accounts.dto';
+
+@Controller('accounts')
+@ApiTags('account-service')
```
Comment (mattheworris) [inline apps/api/src/controllers/accounts.controller.ts:16]: Naming suggestion: How do we feel about `msa-service` for all msa-related api's?

**Review by mattheworris — APPROVED** [review]
Read through the changes, a couple non-blocking comments.

🚢 it

File: apps/api/src/controllers/accounts.controller.ts (hunk 3 above)
Comment (claireolmstead) [reply apps/api/src/controllers/accounts.controller.ts:16]: I changed them to the be the base enpoint name in the next story so this is now "accounts"

**Pull request merged**